kernel/src/engine/arrow_utils.rs

```diff
@@ -1282,20 +1329,33 @@ mod tests {
         result.expect_err("incomplete object");
 
         let input: Vec<Option<&str>> = vec![Some("{}{}")];
+        let result = parse_json_impl(&input.into(), requested_schema.clone()).unwrap();
+        assert_eq!(result.num_rows(), 2);
```

Comment:
this is not correct I don't think? we want two records in one line to produce an error as before.